qa/integration/src/test/resources/features/sso/SsoLoginUsingKeycloakProvider.feature

```diff
@@ -0,0 +1,167 @@
+
+  Scenario: Create external user using the the UserService
+  Login as kapua-sys, create user with externalId
+  Login as external user using jwt credentials
+  Check if user exists
+
+    Given I login as user with name "kapua-sys" and password "kapua-password"
+    And I select account "kapua-sys"
+    Then Account
+      | name      |
+      | test-user |
+    And I configure user service
+      | type    | name                   | value |
+      | boolean | infiniteChildEntities  | true  |
+      | integer | maxNumberChildEntities | 5     |
+    And A generic user
+      | name      | email           | userType | externalId                           |
+      | test-user | test-user@co.co | EXTERNAL | 5698d2ff-d26e-45bf-92df-c3e88da58dad |
+    And I logout
+    And Get access token for user with username "test-user" and password "TestPassword123#"
+    And Create a jwt credential using the access token
+    And Login using a jwt credential
+    Then I logout
+    And I login as user with name "kapua-sys" and password "kapua-password"
+    Then I search for user with name "test-user"
+    And I find user
+      | name      | email           | userType | externalId                           |
+      | test-user | test-user@co.co | EXTERNAL | 5698d2ff-d26e-45bf-92df-c3e88da58dad |
+
+  Scenario: Using normal login using Keycloak credentials
+  Create user using keycloak
+  Login as external user using normal login method
+  Exception should be thrown
+
+    Then Get access token for user with username "test-user" and password "TestPassword123#"
+    And Create a jwt credential using the access token
+    Given Configure the SSO service
+    And Login using a jwt credential
+    Then I logout
+    Given I expect the exception "KapuaAuthenticationException" with the text "*"
+    When I login as user with name "test-user" and password "TestPassword123#"
+    Then An exception was thrown
+
+  Scenario: Login with an existing internal user using the SSO login
+  Login as kapua-sys, create a regular user without externalId
+  Try to login as a keycloak user with same username as previously created user
+  Exception should be thrown
+
+    Given I login as user with name "kapua-sys" and password "kapua-password"
+    And I select account "kapua-sys"
+    Then Account
+      | name      |
+      | test-user |
+    And I configure user service
+      | type    | name                   | value |
+      | boolean | infiniteChildEntities  | true  |
+      | integer | maxNumberChildEntities | 5     |
+    When A generic user
+      | name      | displayName | email           | phoneNumber     | status  | userType |
+      | test-user | Test User   | test-user@co.co | +386 31 323 444 | ENABLED | INTERNAL |
+    And I add credentials
+      | name      | password          | enabled |
+      | test-user | ToManySecrets123# | true    |
+    Then I logout
+    Then Get access token for user with username "test-user" and password "TestPassword123#"
```

Comment:
I think that here the password should be the same as in the "I add credentials" step (just to be coherent with the scenario, even if it shouldn't modify the expected result)

Reply:
I changed the password so it is the same through the whole scenario.